hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -293,21 +317,22 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(final Lon
 
         setAssignedDistributionSetAndTargetUpdateStatus(assignmentStrategy, distributionSetEntity,
                 targetEntitiesIdsChunks);
+    }
 
-        final Map<String, JpaAction> controllerIdsToActions = createActions(targetsWithActionType, targetEntities,
-                assignmentStrategy, distributionSetEntity);
-        // create initial action status when action is created so we remember
-        // the initial running status because we will change the status
-        // of the action itself and with this action status we have a nicer
-        // action history.
-        createActionsStatus(controllerIdsToActions.values(), assignmentStrategy, actionMessage);
+    private DistributionSetAssignmentResult buildAssignmentResult(final JpaDistributionSet distributionSet,
+            final List<JpaTarget> targetEntities, final Map<String, JpaAction> controllerIdsToActions) {
 
-        detachEntitiesAndSendTargetUpdatedEvents(distributionSetEntity, targetEntities, assignmentStrategy);
+        final List<String> targetEntitiesIds = targetEntities.stream().map(Target::getControllerId)
+                .collect(Collectors.toList());
+        final List<Action> alreadyAssignedActions = actionRepository
+                .findAllByDistributionSetIdAndActiveAndTargetControllerIdNotIn(distributionSet.getId(), true,
+                        targetEntitiesIds);
+        final List<Target> alreadyAssignedTargets = alreadyAssignedActions.stream().map(Action::getTarget)
+                .collect(Collectors.toList());
 
-        return new DistributionSetAssignmentResult(distributionSetEntity,
-                targetEntities.stream().map(Target::getControllerId).collect(Collectors.toList()),
-                targetEntities.size(), controllerIDs.size() - targetEntities.size(),
-                Lists.newArrayList(controllerIdsToActions.values()), targetManagement);
+        return new DistributionSetAssignmentResult(distributionSet, alreadyAssignedTargets, 
+                targetManagement.getByControllerID(targetEntitiesIds), Collections.emptyList(),
```

Comment:
why do we need to call `targetManagement.getByControllerID`? why not return the `targetEntities` as is?

Reply:
Targets will be modified and persisted, and we should return the latest version of the targets. Previously, this was handled earlier in `DistributionSetAssignmentResult` through the `targetManagement` field